ibis/expr/types/relations.py

```diff
@@ -1421,19 +1427,21 @@ def order_by(
         │     2 │ B      │     6 │
         └───────┴────────┴───────┘
         """
+        import ibis.selectors as s
+
         sort_keys = []
         for item in util.promote_list(by):
             if isinstance(item, tuple):
                 if len(item) != 2:
-                    raise ValueError(
-                        "Tuple must be of length 2, got {}".format(len(item))
-                    )
-                item = (bind_expr(self, item[0]), item[1])
-                used_tuple_syntax = True
+                    raise ValueError(f"Tuple must be of length 2, got {len(item):d}")
+                sort_keys.append(bind_expr(self, item[0]), item[1])
+            elif isinstance(item, s.Selector):
```

Comment:
It would be nice to condense all this logic in `bind_expr()`, but that requires a larger overhaul.

Reply:
Indeed, we should do that in a follow-up.